akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -168,6 +177,261 @@ private[cluster] final class ClusterHeartbeatSender extends Actor with ActorLogg
 
 }
 
+/*
+ * INTERNAL API
+ *
+ * This actor is responsible for sending the heartbeat messages to
+ * a few other nodes, which will reply and then this actor updates the
+ * failure detector.
+ * 
+ * TODO: The between teams monitoring 
+ */
+private[cluster] final class CrossDcHeartbeatSender() extends Actor with ActorLogging {
+  import ClusterHeartbeatSender._
+  import CrossDcHeartbeatSender._
+
+  val cluster = Cluster(context.system)
+  val verboseHeartbeat = cluster.settings.Debug.VerboseHeartbeatLogging
+  import cluster.{ selfAddress, selfUniqueAddress, scheduler }
+  import cluster.selfDatacenter
+  import cluster.settings._
+  import context.dispatcher
+
+  // TODO this acts like a vector clock, but only a local one.......
+  var localClock: Int = 0
+  def incLocalClock: Int = {
+    localClock += 1
+    localClock
+  }
+  def incLocalClockJson =
+    s"""{"${selfAddress.host.get}:${selfAddress.port.get}": ${incLocalClock}}"""
+  // TODO END OF this acts like a vector clock, but only a local one.......
+
+  /** Allows external actor subscript to status updates by this Actor, used for testing in multi node tests */
+  private[this] var observer: Option[ActorRef] = None
+  /** For inspecting if in active state */
+  var activelyMonitoring = false
+
+  val filterExternalMembers: Member ⇒ Boolean =
+    member ⇒ member.team != cluster.selfDatacenter
+  val crossDcSettings: cluster.settings.CrossDcFailureDetectorSettings = cluster.settings.CrossDcFailureDetectorSettings
+  val crossDcFailureDetector = cluster.crossDcFailureDetector
+
+  val selfHeartbeat = Heartbeat(selfAddress)
+
+  var state: ClusterHeartbeatSenderState = ClusterHeartbeatSenderState(
```

Comment:
I don't think we should use that here, it should be much more simple here (I think it does things we don't want for the cross DC case)